Final version

LizzyRandall opened the pull request:
@michaelharro @ryancarlson Last change that I can see before putting the classes in a pod and putting the mod in mission hub. I added unique ids since they're needed for api calls on filters in Mission hub and would be useful for other implementations using api calls. I also made the sample sub class simpler. Please let me know what you think.

Commit:
Added unique Ids

What I noticed missing at least for mission hub and future uses is a
way to uniquely identify the objects instead of just their
labels/names. I added this in the various methods to do that and had
Alex test again with a simple and smaller subclassed example.

Commit:
Unique ids

Seeing the strings for the mission hub api, I noticed unique ids would
be important to return for a packaged filter. I added functionality to
do this at the various levels so surveys could have an id (say 544)
with the question id (2) with the value for the question.

ryancarlson commented:
This is okay, but makes an assumption that all Identifiers are a numeric type.  It's just as likely that a system could use UUIDs to identify objects which wouldn't work here.

It's not worth making any changes since we don't have this use case in place now; we'll cross that bridge when we come to it.

Commit:
Changed number ids to string ids added logging on example

Strings are needed for mission hub anyway from what I can tell. The
model has a nsnumber for the id but the filter string will be built
using strings, so it’s better to have the unique id as a string. The
NSUUID class can convert the id to a string.

I also added a method printing in the example so the developer could
see the filter returned from the manager.

LizzyRandall commented:
@ryancarlson Thanks for the feedback! It looks like strings are actually what would be best in MH, I was looking at the data model when seeing NSNumber but the filter string passed to the api would be built in a string. I converted all the number ids to strings and if NSUUIDs are used in the future there is a conversion string property, UUIDString.

ryancarlson commented:
:+1:
